clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -286,18 +424,30 @@ cache_simulator_t::check_warmed_up()
 bool
 cache_simulator_t::print_results()
 {
-    std::cerr << "Cache simulation results:\n";
-    for (unsigned int i = 0; i < knobs.num_cores; i++) {
-        print_core(i);
-        if (thread_ever_counts[i] > 0) {
-            std::cerr << "  L1I stats:" << std::endl;
-            icaches[i]->get_stats()->print_stats("    ");
-            std::cerr << "  L1D stats:" << std::endl;
-            dcaches[i]->get_stats()->print_stats("    ");
+    if (is_two_level) {
+        std::cerr << "Cache simulation results:\n";
+        for (unsigned int i = 0; i < knobs.num_cores; i++) {
+            print_core(i);
+            if (thread_ever_counts[i] > 0) {
+                std::cerr << "  L1I stats:" << std::endl;
+                l1_icaches[i]->get_stats()->print_stats("    ");
+                std::cerr << "  L1D stats:" << std::endl;
+                l1_dcaches[i]->get_stats()->print_stats("    ");
+            }
+        }
+        std::cerr << "LL stats:" << std::endl;
+        llcaches[0]->get_stats()->print_stats("    ");
+    }
+    else {
+        std::cerr << "Cache simulation results:\n";
+        for (unsigned int i = 0; i < knobs.num_cores; i++) {
+            print_core(i);
+        }
+        for (auto &caches_it : all_caches) {
```

Comment:
Seems best to have some kind of ordering here: ideally represent the hierarchy, via indentation or sthg?

Reply:
The current order is L1 caches first (ordered by core #), non-L1 non-LLC caches (e.g. L2), and finally LLCs. This works well for a traditional 3-level cache hierarchy. It would be non-trivial to print in an order reflecting more complex hierarchies, so I prefer to defer it to a future change.

Reply:
You mean there's an invariant on all_caches that it is ordered in this manner?  I didn't see a comment explaining that.  I also don't see how your described ordering is accomplished: all_caches is a std::map which is sorted by the key, the cache name.  I can name the caches anything I want, right?

Reply:
This function's implementation has changed.